src/cmd/flux-job.c

```diff
@@ -603,6 +626,175 @@ int cmd_id (optparse_t *p, int argc, char **argv)
     return 0;
 }
 
+struct eventlog_ctx {
+    struct flux_kvs_eventlog *log;
+    optparse_t *p;
+};
+
+/* convert floating point timestamp (UNIX epoch, UTC) to ISO 8601 string,
+ * with microsecond precision
+ */
+static int eventlog_timestr (double timestamp, char *buf, size_t size)
+{
+    time_t sec = timestamp;
+    unsigned long usec = (timestamp - sec)*1E6;
+    struct tm tm;
+
+    if (!gmtime_r (&sec, &tm))
+        return -1;
+    if (strftime (buf, size, "%FT%T", &tm) == 0)
+        return -1;
+    size -= strlen (buf);
+    buf += strlen (buf);
+    if (snprintf (buf, size, ".%.6luZ", usec) >= size)
+        return -1;
+    return 0;
+}
+
+void eventlog_continuation (flux_future_t *f, void *arg)
+{
+    struct eventlog_ctx *ctx = arg;
+    double timestamp;
+    char name[FLUX_KVS_MAX_EVENT_NAME + 1];
+    char context[FLUX_KVS_MAX_EVENT_CONTEXT + 1];
+    char buf[64];
+    const char *s;
+    const char *event;
+
+    if (flux_job_eventlog_lookup_get (f, &s) < 0)
+        log_err_exit ("flux_job_eventlog_lookup_get");
+
+    if (flux_kvs_eventlog_append (ctx->log, s) < 0)
+        log_err_exit ("flux_kvs_eventlog_append");
+
+    while ((event = flux_kvs_eventlog_next (ctx->log))) {
+        if (flux_kvs_event_decode (event, &timestamp, name, sizeof (name),
+                                   context, sizeof (context)) < 0)
+            log_err_exit ("flux_kvs_event_decode");
+
+        if (eventlog_timestr (timestamp, buf, sizeof (buf)) < 0)
+            log_msg_exit ("error converting timestamp to ISO 8601");
+
+        printf ("%s %s%s%s\n", buf, name, *context ? " " : "", context);
+    }
+    fflush (stdout);
+    flux_future_destroy (f);
+}
+
+int cmd_eventlog (optparse_t *p, int argc, char **argv)
+{
+    flux_t *h;
+    int optindex = optparse_option_index (p);
+    flux_jobid_t id;
+    flux_future_t *f;
+    char *endptr;
+    struct eventlog_ctx ctx = {0};
+
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    if (argc - optindex != 1) {
+        optparse_print_usage (p);
+        exit (1);
+    }
+    id = strtoull (argv[optindex++], &endptr, 10);
+    if (*endptr != '\0')
+        log_err_exit ("id requires an unsigned integer argument");
```

Comment:
Switch to `log_msg_exit()` here to suppress a possibly confusing `strerror (errno)` appended to the end of the error, i.e. I'm getting:
```
flux-job: id requires an unsigned integer argument: No such file or directory
```